lib/mysqlDaoQueryHelper.js

```diff
@@ -31,32 +31,35 @@ class MysqlDaoQueryHelper {
 
     /**
      * @param {object} createArgs
+     * @param {boolean} [ignoreOnDuplicateKey] if true, insert will ignore duplicate keys
      * @returns {string}
      */
-    create(createArgs) {
-        const cleanValues = this._cleanAndMapValues(createArgs);
-
-        const sql =
-            Squel.insert().into(this._tableName)
-                .setFields(cleanValues, {dontQuote: true})
-                .toString();
-        return sql;
+    create(createArgs, ignoreOnDuplicateKey) {
+        return this.createBulk([createArgs], ignoreOnDuplicateKey);
     }
 
     /**
      * @param {Array.<object>} createArgsArray
+     * @param {boolean} [ignoreOnDuplicateKey] if true, insert will ignore duplicate keys
      */
-    createBulk(createArgsArray) {
+    createBulk(createArgsArray, ignoreOnDuplicateKey) {
         const cleanValues = [];
         createArgsArray.forEach(createArgs => {
             cleanValues.push(this._cleanAndMapValues(createArgs));
         });
 
-        const sql =
-            Squel.insert().into(this._tableName)
-                .setFieldsRows(cleanValues, {dontQuote: true})
-                .toString();
-        return sql;
+        const sqlBuilder =
+            Squel.useFlavour('mysql').insert().into(this._tableName)
+                .setFieldsRows(cleanValues, {dontQuote: true});
+
+        if (ignoreOnDuplicateKey) {
+            //INSERT ... IGNORE actually ignores ALL errors, not just duplicate key, so best practice is to "update"
+            // a duplicate setting a column to itself
+            const firstField = Object.keys(cleanValues[0])[0];
+            sqlBuilder.onDupUpdate(firstField, firstField, {dontQuote: true})
```

Comment:
The SQL it generates is fine. `col1 = col1` basically well set it to the row value, and MySQL is smart enough to know it doesn't even need to touch the row.